### ISSUE TITLE
PID file directory not found

### ISSUE THREAD

**Issue by ZlatyChlapec** [troglobit/inadyn#284]
When you pass custom pid file directory with `-P` it doesn't work in case some of directories on path `/usr/var/run/` are missing.

After I created directories `var` and `run` pid file was created without any problems in my custom path.

**Comment by troglobit** [comment]
That is correct, In-a-dyn can not be made responsible for creating system directories, that is up to the operator or distribution packaging In-a-dyn.

**Comment by ZlatyChlapec** [comment]
I agree on that. But should it be still expecting those directories to exist when a specified that I want to use different directories for that? That's what got me confused. Because I have set that I want to use some directories for storage of pid files which existed. But In-a-dyn was still not able to create pid file in them until I didn't create default directories as well.

**Comment by troglobit** [comment]
Sorry, I don't understand.  You start with `inadyn -P /usr/var/run/inadyn.pid`, which fails and then it works when you create `/usr/var` and `/usr/var/run` ... what am I missing here?  Could you perhaps tell me exactly what you did, and what you expected in each step?

**Comment by ZlatyChlapec** [comment]
Oh I am sorry. I didn't made that clear.  
For example when I start with `inadyn -P /tmp/pids/inadyn.pid` unless I have all `/usr/var/run` directories created it'll end up with missing directory error. Even tho ` /tmp/pids` exists.

**Comment by troglobit** [comment]
Aha, that makes more sense. It should definitely not behave like that.  What's the exact error message you get?

**Comment by ZlatyChlapec** [comment]
This is the message `user.warn inadyn[5489]: Failed creating pidfile: No such file or directory`. It's only warning but openRC was not able to work with In-a-dyn because of that.  
I tried to look into code and fix it but I don't have any experience with C ;(.

**Comment by troglobit** [comment]
Thank you for taking the time to clarify this, I'll have a look!  If it's a bug I'll push something to master within a day or so with a commit message that will close this issue :)

**Comment by troglobit** [comment]
Not really sure I fixed your issue.  But it turns out Inadyn *does* indeed try to create directories for both the PID file and the cache file(s) (used to cache the IP address sent to the DDNS provider, to protect against updating too often so you don't get locked out from your DDNS account).

I've made the code paths in `parse_privs()` and `os_check_perms()` a bit more forgiving and made sure they log warnings for missing system directories.  I hope this fixes your problem.
